server/src/main/java/org/opensearch/indices/IndicesRequestCache.java

```diff
@@ -136,6 +137,14 @@ public final class IndicesRequestCache implements RemovalListener<IndicesRequest
                 .setRemovalListener(this)
                 .setMaxSizeInBytes(sizeInBytes) // for backward compatibility
                 .setExpireAfterAccess(expire) // for backward compatibility
+                .setCachedResultParser((bytesReference) -> {
+                    try {
+                        return CachedQueryResult.getPolicyValues(bytesReference);
+                    } catch (IOException e) {
+                        // Set took time to -1, which will always be rejected by the policy.
+                        return new CachedQueryResult.PolicyValues(-1);
```

Comment:
For this to work you need to avoid users from setting the new value of `TIERED_SPILLOVER_DISK_TOOK_TIME_THRESHOLD` as -1  which can be done today. Can you also add a test for both the setting related validation and this case